cs130_front/src/components/views/ClassesPage/ClassesPage.js

```diff
@@ -11,42 +12,155 @@ import SearchBar from '../../Search/SearchBar';
+      const groupId = data[key]["id"];
+      const currentGroup = {name: name, groupId: groupId};
+      if (!(data[key]["course_name"] in allGroups)){
+        allGroups[data[key]["course_name"]] = [];
+      }
+      allGroups[data[key]["course_name"]].push(currentGroup);
+    });
+    setAllGroups(allGroups);
+    return allGroups;
+  }
+
+  function handleBuddiesListingsResponse(data, allGroups){
+
+    const i = [{name: "Wow Squad", groupId: 123}, {name: "CA Squad", groupId: 124}, {name: "DM Squad", groupId: 125}]
+
+    const classes = []
+    Object.keys(data).forEach(function(key) {
+      const studyBuddies = [];
+      const userDict = {};
+      Object.keys(data[key]["StudyBuddies"]).forEach(function(key2) {
+        const name = data[key]["StudyBuddies"][key2]["first_name"] + " " + data[key]["StudyBuddies"][key2]["last_name"];
+        const id = data[key]["StudyBuddies"][key2]["u_id"];
+        const school = data[key]["StudyBuddies"][key2]["school_name"];
+        const email = data[key]["StudyBuddies"][key2]["u_email"];
+        const discord = "None";
+        if (!(id == parseInt(userId))){
+          studyBuddies.push({name: name, school: school, id: id, discord: discord, email: email});
+        }
+        userDict[id] = {name: name, school: school};
+      });
+
+      const courseId = key;
+      const name = data[key]["CourseName"];
+      const listings = [];
+      console.log(data[key]["Listings"]);
+      Object.keys(data[key]["Listings"]).forEach(function(key2) {
+        const poster = data[key]["Listings"][key2]["poster"];
+        const name = userDict[poster].name;
+        const school = userDict[data[key]["Listings"][key2]["poster"]].school;
+        const description = data[key]["Listings"][key2]["text_description"];
+        const group_id = data[key]["Listings"][key2]["group_id"];
+        const group_name = data[key]["Listings"][key2]["group_name"];
+        if (!(data[key]["Listings"][key2]["poster"] == parseInt(userId))){
+          listings.push({poster: poster, name: name, school: school, description: description, group_id: group_id, group_name: group_name});
+        }
+      });
+      classes.push({name:name, courseId: courseId, listings: listings, studyBuddies: studyBuddies});
+    });
+    setClasses(classes);
+    setMainTitle(classes[0].name);
+    setMainClassId(classes[0].courseId);
+    setMainListingsDefault(classes[0].listings);
+    setMainStudyBuddies(classes[0].studyBuddies);
+    setMainGroups(allGroups[classes[0].name]);
+  }
+
+  function createListing(listing) {
+    const body = {
+      poster: parseInt(listing.poster),
+      course_id: parseInt(listing.course_id),
+      text_description: listing.text_description,
+      course_name: listing.course_name
+    }
+
+    if (listing.group_id != 0){
+      body["group_id"] = listing.group_id;
+      body["group_name"] = listing.group_name;
+    }
+
+    axios.post(`${COURSE_SERVER}/addListing?u_id=${userId}`, body, config);
```

Comment:
in order to reload when the listing is added: 

```
    axios.post(`${COURSE_SERVER}/addListing?u_id=${userId}`, body, config).then(response => {
      return axios.all([getCourses(), getUser()]);
    }).then( axios.spread((data1, data2) => {
      const allGroups = handleUserResponse(data2.data.groups);
      handleBuddiesListingsResponse(data1.data, allGroups);
      setAddedListing(listing);
    }));
```
The setAddedListing will trigger the useEffect above to choose the correct class that had the listing added and shows the listing that was added immediately.